bioptim/interfaces/SolverOptions.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass
+class SolverOptions(ABC):
```

Comment:
Please do not use truncated wordings for the methods (`set_convergence_tolerance`, `set_maximum_iterations`). it does not take long to write (and that this autocompleted anyway, and is much more verbose